management/src/clusterm/manager/api.go

```diff
@@ -72,11 +73,12 @@ func (m *Manager) apiLoop(errCh chan error) {
 	}
 }
 
-type postCallback func(tagsOrAddrs []string, sanitizedExtraVars string) error
+type postCallback func(tagsOrAddrs []string, flags ActionFlags) error
```

Comment:
cool, yeah I think using `*APIRequest` here will make more sense. 

@vvb , are you saying you want to tackle #99 as well as part of this? I am completely fine (and would prefer) if we take care of extraVars separately than expanding the scope of this PR. You can take over that one after this if you like.